language/move-core/types/src/vm_status.rs

```diff
@@ -235,9 +235,89 @@ pub fn convert_prologue_runtime_error(status: VMStatus) -> VMStatus {
             };
             VMStatus::Error(new_major_status)
         }
-        status @ VMStatus::ExecutionFailure { .. }
-        | status @ VMStatus::Error(_)
+        status @ VMStatus::ExecutionFailure { .. } | status @ VMStatus::Executed => status,
+        VMStatus::Error(code) => {
+            debug_assert!(false, "Unexpected prologue error: {:?}", code);
+            VMStatus::Error(StatusCode::UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION)
+        }
+    }
+}
+
+/// Checks for only move aborts, execution failures, and successful execution.
+/// Any other errors are mapped to the invariant violation
+/// `UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION`
+pub fn convert_normal_epilogue_error(status: VMStatus) -> VMStatus {
```

Comment:
It feels like these convert functions don't belong to Move VM Status and can be migrated over to LibraVm.

Reply:
Thanks for the suggestion! Cleaned things up a bit and made logging easier to boot!